packages/react-hooks/src/useControllableValue.ts

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { useConst } from './useConst';
+
+export type ChangeCallback<TElement extends HTMLElement, TValue> = (
+  ev: React.FormEvent<TElement> | undefined,
+  newValue: TValue | undefined,
+) => void;
+
+/**
+ * Hook to manage a value that could be either controlled or uncontrolled, such as a checked state or
+ * text box string.
+ * @param controlledValue- The controlled value passed in the props. This value will always be used if provided, and the
+ * internal state will be updated to reflect it.
+ * @param defaultUncontrolledValue- Initial value for the internal state in the uncontrolled case.
+ * @see https://reactjs.org/docs/uncontrolled-components.html
+ */
+export function useControllableValue<TValue, TElement extends HTMLElement>(
+  controlledValue: TValue | undefined,
+  defaultUncontrolledValue: TValue | undefined,
+): Readonly<[TValue | undefined, (newValue: TValue | undefined) => void]>;
+export function useControllableValue<
+  TValue,
+  TElement extends HTMLElement,
+  TCallback extends ChangeCallback<TElement, TValue> | undefined
+>(
+  controlledValue: TValue | undefined,
+  defaultUncontrolledValue: TValue | undefined,
+  onChange: TCallback,
+): Readonly<[TValue | undefined, (newValue: TValue | undefined, ev?: React.FormEvent<TElement>) => void]>;
+export function useControllableValue<
+  TValue,
+  TElement extends HTMLElement,
+  TCallback extends ChangeCallback<TElement, TValue> | undefined
+>(controlledValue: TValue | undefined, defaultUncontrolledValue: TValue | undefined, onChange?: TCallback) {
+  const [value, setValue] = React.useState<TValue | undefined>(defaultUncontrolledValue);
+  const isControlled = useConst<boolean>(!!controlledValue);
```

Comment:
This won't have the desired result for empty strings or 0. Should do either `controlledValue !== undefined` or `controlledValue != null` (depending on how we want to handle null).

Reply:
Yeah, the tests caught me on this one in my last run. Already fixed. 👍